### PR TITLE
pinocchio 2.6.10 (new formula)

### PR DESCRIPTION
**Pull request by jcarpent**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/pinocchio.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  depends_on "eigenpy"
+  depends_on "hpp-fcl"
+  depends_on "python@3.10"
+  depends_on "urdfdom"
+
+  def python3
+    deps.map(&:to_formula)
+        .find { |f| f.name.match?(/^python@\d\.\d+$/) }
+        .opt_libexec/"bin/python"
+  end
```
Comment (SMillerDev) [inline Formula/pinocchio.rb:22]: This is only ever going to return `Formula["python@3.10"].opt_libexec/"bin/python"` so you can just use that

Comment (jcarpent) [reply Formula/pinocchio.rb:22]: Thanks @SMillerDev for the quick feedback.

**Comment by jcarpent** [comment]
@SMillerDev This PR is ready for a second review pass

File: Formula/pinocchio.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  def install
+    if build.head?
+      system "git", "submodule", "update", "--init"
+      system "git", "pull", "--unshallow", "--tags"
+    end
+
+    ENV.prepend_path "PYTHONPATH", Formula["eigenpy"].opt_prefix/Language::Python.site_packages(python3)
+    ENV.prepend_path "Eigen3_DIR", Formula["eigen"].opt_share/"eigen3/cmake"
```
Comment (SMillerDev) [inline Formula/pinocchio.rb:29]: Can't cmake pick up on these otherwise?

File: Formula/pinocchio.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  depends_on "boost"
+  depends_on "eigen"
+  depends_on "eigenpy"
+  depends_on "hpp-fcl"
+  depends_on "python@3.10"
+  depends_on "urdfdom"
+
+  def python3
+    Formula["python@3.10"].opt_libexec/"bin/python"
+  end
```
Comment (SMillerDev) [inline Formula/pinocchio.rb:20]: Since it's only needed once this can be used directly instead of the method.

**Comment by jcarpent** [comment]
@SMillerDev I've handled your remarks. Thanks a lot.

**Review by chenrui333 — DISMISSED** [review]

**Comment by chenrui333** [comment]
LGTM, might need a rebase to re-generate the bottles (bottles might expire)

**Review by chenrui333 — DISMISSED** [review]

File: Formula/pinocchio.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                    "-DPYTHON_EXECUTABLE=#{Formula["python@3.10"].opt_libexec/"bin/python"}",
+                    "-DBUILD_UNIT_TESTS=OFF",
+                    "-DBUILD_WITH_COLLISION_SUPPORT=ON",
+                    *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    system python3, "-c", <<~EOS
```
Comment (SMillerDev) [inline Formula/pinocchio.rb:38]:
```suggestion
    system bin/"python3.10", "-c", <<~EOS
```

Comment (chenrui333) [reply Formula/pinocchio.rb:38]: Didn't python3 get defined in line 18 - 20?

**Comment by jcarpent** [comment]
Thank you all for the excellent work you are doing with HomeBrew.
@SMillerDev Do you know when this PR will be merged?

File: Formula/pinocchio.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+                    "-DBUILD_UNIT_TESTS=OFF",
+                    "-DBUILD_WITH_COLLISION_SUPPORT=ON",
+                    *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    python_exe = Formula["python@3.10"].opt_libexec/"bin/python"
+    system python_exe, "-c", <<~EOS
```
Comment (SMillerDev) [inline Formula/pinocchio.rb:35]: Is it just a python library? Or is that the only thing we're testing?

Comment (jcarpent) [reply Formula/pinocchio.rb:35]: This is a C++ lib with Python bindings.
We are only testing the Python bindings, which allows to certify the whole library is well compiled and installed.

**Review by SMillerDev — APPROVED** [review]

**Comment by jcarpent** [comment]
Super. Thanks a lot, @SMillerDev for the feedback.

**Pull request closed without merging**